core/src/main/java/org/apache/iceberg/GenericPartitionFieldSummary.java

```diff
@@ -136,8 +144,10 @@ public Object get(int i) {
       case 0:
         return containsNull;
       case 1:
-        return lowerBound();
+        return containsNaN;
```

Comment:
have we made sure changing the position is backwards compatible?

Reply:
This is a good point, I forgot that primitive boolean will be default to false which will impact correctness when we use this in evaluator. I'll make this nullable. Thanks for pointing this out!

Reply:
The position change is compatible. All Iceberg metadata files conform to the Iceberg data file spec, so we can reorder columns if we choose.

Reply:
Ah sorry I misread the original question... Thanks Ryan for help with answering it!